lib/Prometheus/Servant/Internal.hs

```diff
@@ -225,11 +225,30 @@ instance HasEndpoint Raw where
   enumerateEndpoints _ = [Endpoint [] "RAW"]
 
 instance HasEndpoint (sub :: Type) => HasEndpoint (CaptureAll (h :: Symbol) a :> sub) where
+  getEndpoint _ req =
+    case pathInfo req of
+      [] -> Nothing
+      _ -> do
+        Endpoint{..} <- getEndpoint (Proxy :: Proxy sub) req{pathInfo = []}
+        pure $ Endpoint ("*" : ePathSegments) eMethod
+
+  enumerateEndpoints _ = do
+    let qualify Endpoint{..} = Endpoint ("*" : ePathSegments) eMethod
+    map qualify $ enumerateEndpoints (Proxy :: Proxy sub)
+
+instance HasEndpoint (sub :: Type) => HasEndpoint (BasicAuth (realm :: Symbol) a :> sub) where
   getEndpoint _ = getEndpoint (Proxy :: Proxy sub)
 
   enumerateEndpoints _ = enumerateEndpoints (Proxy :: Proxy sub)
 
-instance HasEndpoint (sub :: Type) => HasEndpoint (BasicAuth (realm :: Symbol) a :> sub) where
+#if MIN_VERSION_servant(0,2,0)
+instance HasEndpoint (sub :: Type) => HasEndpoint (WithResource a :> sub) where
   getEndpoint _ = getEndpoint (Proxy :: Proxy sub)
 
   enumerateEndpoints _ = enumerateEndpoints (Proxy :: Proxy sub)
+
+instance HasEndpoint RawM where
+  getEndpoint _ _ = Just (Endpoint [] "RAW")
+
+  enumerateEndpoints _ = [Endpoint [] "RAW"]
+#endif
```

Comment:
@worm2fed I think the version bounds is correct here, the version has to be greater or equal than the specified version, I know there is no `0.2.0` and only a `0.2` but the macro requires 3 arguments.

https://ghc.gitlab.haskell.org/ghc/doc/users_guide/phases.html#standard-cpp-macros
>MIN_VERSION_pkgname(x,y,z)
>
>    This macro is available starting GHC 8.0. It is defined for every exposed package. This macro is provided for convenience to write CPP conditionals testing if a package version is x.y.z or later. It is identical in behavior to the MIN_VERSION_pkgname macros that Cabal defines.

Reply:
Sorry I see what you meant its supposed to be 0.20 not 0.2

Reply:
I have fixed this to be 

```
#if MIN_VERSION_servant(0,20,0)
```